File: render-pdf.mjs
import puppeteer from "puppeteer";
import path from "path";
import { fileURLToPath } from "url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const browser = await puppeteer.launch({ headless: true });
const page = await browser.newPage();
await page.goto(`file://${__dirname}/report.html`);
await page.pdf({ path: "report.pdf" });

await browser.close();

File: render_pdf.py
from playwright.sync_api import sync_playwright
import os


def render_pdf():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        page.goto(f"file://{os.path.abspath("report.html")}")
        page.emulate_media(media="screen")
        page.pdf(path="report-py.pdf")
        browser.close()


print(f"module name: {__name__}")
